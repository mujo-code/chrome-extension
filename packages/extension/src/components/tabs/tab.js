import { BodyS, useTheme, Box } from '@mujo/ui'
import React, { useState } from 'react'

export const Tab = ({ label, selected, onClick }) => {
  const [hovered, setHovered] = useState(false)
  const { highlight } = useTheme()
  const yPosition = hovered ? 4 : 8
  const css = {
    marginLeft: '-8px',
    marginBottom: '-8px',
    left: '50%',
    bottom: 0,
    transition: 'transform 0.2s ease-in-out',
    transform: `translateY(${selected ? 0 : yPosition}px) scale(${
      selected ? 1 : 0.8
    })`,
  }
  return (
    <Box
      position="relative"
      paddingLeft="m"
      paddingRight="m"
      whiteSpace="nowrap"
      overflow="hidden"
      cursor="pointer"
      onClick={onClick}
      onMouseEnter={() => setHovered(true)}
      onMouseLeave={() => setHovered(false)}
    >
      <Box
        position="absolute"
        height="m"
        width="m"
        borderRadius="m"
        css={css}
        backgroundColor={highlight}
      />
      <BodyS>{label}</BodyS>
    </Box>
  )
}
