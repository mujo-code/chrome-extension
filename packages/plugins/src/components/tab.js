import { Ingress } from '@mujo/ingress'
import React, { useEffect, useContext } from 'react'
import { context } from './plugin-provider'

export const Tab = ({ name, children }) => {
  const { constants, tabs, env } = useContext(context)
  const { pushTab, removeTab, currentTab } = tabs

  useEffect(() => {
    if (env !== 'ntp') return () => {}
    pushTab(name)
    return () => removeTab(name)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [name])
  // only supported on ntp page
  if (env !== 'ntp') return null
  if (name !== currentTab) return null
  return <Ingress target={constants.TABS_TARGET}>{children}</Ingress>
}
