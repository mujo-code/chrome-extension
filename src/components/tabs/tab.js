import { Box } from '@mujo/box'
import React, { useState } from 'react'
import { useTheme } from '../../hooks/use-theme'
import { BodyS } from '../fonts'

export const Tab = ({ label, selected, onClick }) => {
  const [hovered, setHovered] = useState(false)
  const { highlight } = useTheme()
  const yPosition = hovered ? 4 : 8
  const css = {
    marginLeft: '-8px',
    marginBottom: '-8px',
    left: '50%',
    bottom: 0,
    transition: 'transform 0.2s ease-in-out',
    transform: `translateY(${selected ? 0 : yPosition}px) scale(${
      selected ? 1 : 0.8
    })`,
  }
  return (
    <Box
      position="relative"
      paddingLeft="m"
      paddingRight="m"
      whiteSpace="nowrap"
      overflow="hidden"
      cursor="pointer"
      onClick={onClick}
      onMouseEnter={() => setHovered(true)}
      onMouseLeave={() => setHovered(false)}
    >
      <Box
        position="absolute"
        height="m"
        width="m"
        borderRadius="m"
        css={css}
        backgroundColor={highlight}
      />
      <BodyS>{label}</BodyS>
    </Box>
  )
}
