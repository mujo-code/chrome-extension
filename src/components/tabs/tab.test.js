import React from 'react'
import { create } from 'react-test-renderer'
import { Tab } from './tab'

test('Tab component matches snapshot', () => {
  const tree = create(<Tab label="foo" />).toJSON()
  expect(tree).toMatchSnapshot()
})
