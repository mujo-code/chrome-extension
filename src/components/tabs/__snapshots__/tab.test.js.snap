// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component matches snapshot 1`] = `
<div
  className="css-17xbw2"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  overflow="hidden"
  position="relative"
>
  <div
    backgroundColor="saltBox"
    className="css-u4k9dx"
    height="m"
    position="absolute"
    width="m"
  />
  <p
    className="css-1hhdzfd"
    color="color"
  >
    foo
  </p>
</div>
`;
